pkg/controller/controllerutils/controller_errors.go

```diff
@@ -0,0 +1,20 @@
+package controllerutils
```

Comment:
honestly it probably just makes sense to move these to the utils package

Reply:
doesn't make sense for there to be a difference

Reply:
these are pretty scoped to controller logic though, right? we're never going to have a "user error" that gets created from other packages.

I like having this be within controller because going forward we'll likely want more controller-specific utils anyway

Reply:
same for upsert and some of the utils in the utils package

Reply:
touche

Reply:
moved